src/mediafileattribute.cpp

```diff
@@ -420,6 +420,37 @@ MediaProperties::MediaProperties()
 {
 }
 
+MediaProperties::MediaProperties(const std::string& deserialize)
+{
+    CacheableReader r(deserialize);
+    r.unserializebyte(shortformat);
+    r.unserializeu32(width);
+    r.unserializeu32(height);
+    r.unserializeu32(fps);
+    r.unserializeu32(containerid);
+    r.unserializeu32(videocodecid);
+    r.unserializeu32(audiocodecid);
+    r.unserializebool(is_VFR);
+    r.unserializebool(no_audio);
+}
```

Comment:
I'm not sure if there will be a case where we wan this to receive `string *deserialize` ?  and then do `r.eraseused(deserialize);` by the end

Reply:
Good point.  There isn't currently for this particular class, and if we do need it in the future then we can easily add it unconditionally, or with a boolean parameter on the function.   However I think in future we could move to the serialization/deserialization functions receiving CacheableWriter/Reader objects instead of strings, which would mean we could process member objects without any need for eraseused() at all, which would improve efficiency - but that would be too much to introduce in this PR.